diff --git a/node_modules/react-native-markdown-display/src/lib/renderRules.js b/node_modules/react-native-markdown-display/src/lib/renderRules.js
index 6f2ed8d..02e1ab4 100644
--- a/node_modules/react-native-markdown-display/src/lib/renderRules.js
+++ b/node_modules/react-native-markdown-display/src/lib/renderRules.js
@@ -1,18 +1,20 @@
-import React from 'react';
 import {
-  Text,
-  TouchableWithoutFeedback,
-  View,
+  ActivityIndicator,
+  Image,
   Platform,
   StyleSheet,
+  Text,
+  TouchableWithoutFeedback,
+  View
 } from 'react-native';
-import FitImage from 'react-native-fit-image';
 
-import openUrl from './util/openUrl';
 import hasParents from './util/hasParents';
+import openUrl from './util/openUrl';
 
 import textStyleProps from './data/textStyleProps';
 
+import { useState } from 'react';
+
 const renderRules = {
   // when unknown elements are introduced, so it wont break
   unknown: (node, children, parent, styles) => null,
@@ -282,21 +284,17 @@ const renderRules = {
       return null;
     }
 
-    const imageProps = {
-      indicator: true,
-      key: node.key,
-      style: styles._VIEW_SAFE_image,
-      source: {
-        uri: show === true ? src : `${defaultImageHandler}${src}`,
-      },
-    };
+    const uri = show === true ? src : `${defaultImageHandler}${src}`;
 
-    if (alt) {
-      imageProps.accessible = true;
-      imageProps.accessibilityLabel = alt;
-    }
-
-    return <FitImage {...imageProps} />;
+    // Use the OriginalSizeImage component to maintain original dimensions
+    return (
+      <OriginalSizeImage 
+        key={node.key}
+        uri={uri}
+        alt={alt}
+        style={styles._VIEW_SAFE_image}
+      />
+    );
   },
 
   // Text Output
@@ -344,4 +342,48 @@ const renderRules = {
   ),
 };
 
+const OriginalSizeImage = ({ uri, alt, style }) => {
+
+  const [layoutDimensions, setLayoutDimensions] = useState(parseImageDimensions(alt));
+  
+  const imageProps = {
+    indicator: true,
+    style: {
+      ...style,
+      width: layoutDimensions.width,  
+      height: layoutDimensions.height, 
+      resizeMode: 'contain', 
+      alignSelf: 'flex-start',
+    },
+    source: { uri },
+    onLayout: (event) => {
+      const { width, height } = event.nativeEvent.layout;
+      setLayoutDimensions({ width, height });
+    }
+  };
+
+  if (alt) {
+    imageProps.accessible = true;
+    imageProps.accessibilityLabel = alt;
+  }
+
+  if(layoutDimensions.width === undefined || layoutDimensions.height === undefined) {
+    return <ActivityIndicator size={20} style = {{marginHorizontal: 10}}/>
+  }
+
+  return <View style = {{width: layoutDimensions.width, height: layoutDimensions.height, marginHorizontal: 10}}>
+            <Image {...imageProps} />
+         </View>;
+};
+
+function parseImageDimensions(imageDimensions) {
+
+  const [width, height] = imageDimensions.split(",")
+
+  return {
+    width: width ?? undefined,
+    height: height ?? undefined
+  }
+}
+
 export default renderRules;
